source/src/main/java/com/kezong/fataar/RTransform.java

```diff
@@ -0,0 +1,202 @@
+package com.kezong.fataar;
+
+import com.android.build.api.transform.DirectoryInput;
+import com.android.build.api.transform.Format;
+import com.android.build.api.transform.QualifiedContent;
+import com.android.build.api.transform.Status;
+import com.android.build.api.transform.Transform;
+import com.android.build.api.transform.TransformInput;
+import com.android.build.api.transform.TransformInvocation;
+import com.android.build.api.transform.TransformOutputProvider;
+import com.android.build.gradle.internal.pipeline.TransformManager;
+import com.google.common.collect.ImmutableSet;
+
+import org.gradle.api.Project;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import javassist.CannotCompileException;
+import javassist.ClassPool;
+import javassist.CtClass;
+import javassist.NotFoundException;
+import javassist.bytecode.ClassFile;
+import javassist.bytecode.ConstPool;
+import kotlin.io.FilesKt;
+
+/**
+ * com.sdk.R
+ * |-- com.lib1.R
+ * |-- com.lib2.R
+ * <p>
+ * rename com.lib1.R and com.lib2.R to com.sdk.R
+ */
+public class RTransform extends Transform {
+
+    private static final int CPU_COUNT = Runtime.getRuntime().availableProcessors();
+
+    private final ExecutorService executor = Executors.newFixedThreadPool(CPU_COUNT + 1);
```

Comment:
In Gradle there's [Worker API](https://docs.gradle.org/nightly/userguide/worker_api.html) for splitting task action into smaller actions, better to use it instead